plugins/nf-validation/src/main/nextflow/validation/SamplesheetConverter.groovy

```diff
@@ -262,26 +262,47 @@ class SamplesheetConverter {
             return result
         }
 
-        // Check and convert integer values
-        else if(type == "integer" || type == "number") {
+        // Convert number values
+        else if(type == "number") {
+            try {
+                def int result = input as int
+                return result
+            }
+            catch (NumberFormatException e) {
+                log.debug("Could not convert ${input} to an integer. Trying to convert to a float.")
+            }
+
+            try {
+                def float result = input as float
+                return result
+            }
+            catch (NumberFormatException e) {
+                log.debug("Could not convert ${input} to a float. Trying to convert to a double.")
+            }
+            
+            def double result = input as double
```

Comment:
Should we catch an error here too? And maybe throw a more explanatory message saying that the input is not a value. Although this will in theory be detected by the validation before the channel conversion.

Reply:
Yeah that was my reasoning too, the values should be compatible when they are validated so they shouldn't really fail